noxfile.py

```diff
@@ -0,0 +1,279 @@
+@nox.session
+def tests(session: "nox.session") -> None:
+    session_id = "tests_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def tests_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "tests_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def tests_version(session: "nox.session", version: str) -> None:
+    session.install(
+        "jinja2",
+        "lxml",
+        "pygments==2.7.4",
+        "pytest",
+        "cmake",
+        "yaxmldiff",
+    )
+    coverage_args = []
+    if os.environ.get("USE_COVERAGE") == "true":
+        session.install("pytest-cov")
+        coverage_args = ["--cov=gcovr", "--cov-branch"]
+    session.install("-e", ".")
+    set_environment(session, version)
+    session.log("Print tool versions")
+    session.run("python", "--version")
+    session.run(session.env["CC"], "--version", external=True)
+    session.run(session.env["CXX"], "--version", external=True)
+    session.run(
+        session.env["GCOV"].replace(" gcov", "")
+        if "llvm-cov" in session.env["GCOV"]
+        else session.env["GCOV"],
+        "--version",
+        external=True,
+    )
+
+    session.run("bash", "-c", "cd gcovr/tests && make --silent clean", external=True)
+    args = ["-m", "pytest"]
+    args += coverage_args
+    args += session.posargs
+    if "--" not in args:
+        args += ["--", "gcovr", "doc/examples"]
+    session.run("python", *args)
+
+
+@nox.session
+def build_wheel(session: "nox.session") -> None:
+    session.install("wheel", "twine")
+    session.run("python", "setup.py", "sdist", "bdist_wheel")
+    session.run("twine", "check", "dist/*", external=True)
+
+
+@nox.session
+def upload_wheel(session: "nox.session") -> None:
+    session.install("twine")
+    session.run("twine", "upload", "dist/*", external=True)
+
+
+def docker_container_id(version: str) -> None:
+    return "gcovr-qa-{}".format(version)
+
+
+@nox.session
+def docker_qa_build(session: "nox.session") -> None:
+    session_id = "docker_qa__build_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def docker_qa_build_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "docker_qa_build_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def docker_qa_build_version(session: "nox.session", version: str) -> None:
+    set_environment(session, version, False)
+    session.run(
+        "bash",
+        "-c",
+        " ".join(
```

Comment:
Doesn't need bash. The shell interpolation for the `CC` and `CXX` variables can be done in Python.

Reply:
I'll change this.